### PR TITLE
Hubble-Relay: proxy context from originating client

### PR DESCRIPTION
**Pull request by nathanjsweet**
The context that comes with the GRPC request has valuable
metadata information in it that is useful for things like
authentication and authorization for downstream servers.

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

```release-note
Hubble-relay now proxies the GRPC context to its servers.
```

### CODE REVIEW

**Comment by michi-covalent** [comment]
the change looks pretty innocent, but this failure might be related:

https://travis-ci.com/github/cilium/cilium/jobs/371489186#L6270

**Review by aanm — APPROVED** [review]

File: pkg/hubble/relay/observer/server.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ sortedFlowsLoop:
 // ServerStatus implements observerpb.ObserverServer.ServerStatus by aggregating
 // the ServerStatus answer of all hubble peers.
 func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusRequest) (*observerpb.ServerStatusResponse, error) {
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
```
Comment (glibsm) [inline pkg/hubble/relay/observer/server.go]: shadow var here is not great

**Review by michi-covalent — APPROVED** [review]

**Comment by michi-covalent** [comment]
retest-net-next

**Comment by michi-covalent** [comment]
retest-4.9

**Comment by michi-covalent** [comment]
retest-4.19

**Comment by michi-covalent** [comment]
retest-runtime

**Comment by nathanjsweet** [comment]
We need to wait for mTLS to get merged before testing this becomes feasible.

**Comment by aanm** [comment]
@nathanjsweet the mtls PR was merged. are we good to go?

**Comment by tklauser** [comment]
Note for future backports: I attempted to backport this to `v1.8` but hit a segfault in the privileged tests, see https://github.com/cilium/cilium/pull/13951#issuecomment-724008296, thus dropped it from #13951

**Comment by Rolinh** [comment]
@tklauser This is not a privileged test but in any case, I believe that pulling in #13206 would fix the segfault. I marked it for for 1.8 backport.

**Pull request merged**